client/core/core.go

```diff
@@ -768,6 +769,9 @@ type Config struct {
 	DBPath string
 	// Net is the current network.
 	Net dex.Network
+	// Logger is the Core's logger and is also used to create the sub-loggers
+	// for the asset backends.
+	Logger dex.Logger
```

Comment:
Wondering if constructor should ensure this is not a nil interface.  dhill is going to upgrade, but the Core API has no breaking changes.  Could be surprising to panic on first log attempt.

The fact that we've gone from the default of a "disabled" logger that does nothing to needing to set it is bittersweet since it won't let us accidentally neglect the logger, but it could panic at inconvenient times if the type's logging is only used in rare error cases (e.g. client's EpochQueue).